core/extensions/impl/storage_extension.cpp

```diff
@@ -3,99 +3,185 @@
+          "memory");
+    }
+    return data_ptr;
   }
 
-  uint32_t StorageExtension::ext_exists_child_storage(
-      const uint8_t *storage_key_data, uint32_t storage_key_length,
-      const uint8_t *key_data, uint32_t key_length) {
-    std::terminate();
+  runtime::SizeType StorageExtension::ext_get_storage_into(
+      runtime::WasmPointer key_data, runtime::SizeType key_length,
+      runtime::WasmPointer value_data, runtime::SizeType value_length,
+      runtime::SizeType value_offset) {
+    auto key = memory_->loadN(key_data, key_length);
+    auto data = get(key, value_offset, value_length);
+    if (not data) {
+      return runtime::WasmMemory::kMaxMemorySize;
+    }
+    memory_->storeBuffer(value_data, data.value());
+    return data.value().size();
   }
 
-  uint8_t *StorageExtension::ext_get_allocated_child_storage(
-      const uint8_t *storage_key_data, uint32_t storage_key_length,
-      const uint8_t *key_data, uint32_t key_length, uint32_t *written) {
-    std::terminate();
+  void StorageExtension::ext_set_storage(const runtime::WasmPointer key_data,
+                                         runtime::SizeType key_length,
+                                         const runtime::WasmPointer value_data,
+                                         runtime::SizeType value_length) {
+    auto key = memory_->loadN(key_data, key_length);
+    auto value = memory_->loadN(value_data, value_length);
+    auto put_result = db_->put(key, value);
+    if (not put_result) {
+      logger_->error(
+          "ext_set_storage failed, due to fail in trie db with reason: {}",
+          put_result.error().message());
+    }
   }
 
-  uint8_t *StorageExtension::ext_get_allocated_storage(const uint8_t *key_data,
-                                                       uint32_t key_length,
-                                                       uint32_t *written) {
+  // -------------------------Trie operations--------------------------
+
+  void StorageExtension::ext_blake2_256_enumerated_trie_root(
+      runtime::WasmPointer values_data, runtime::WasmPointer lens_data,
+      runtime::SizeType lens_length, runtime::WasmPointer result) {
+    // TODO(Akvinikym) PRE-54 11.03.19: implement, when Merkle Trie is ready
+    logger_->error("Unimplemented");
     std::terminate();
   }
 
-  uint32_t StorageExtension::ext_get_child_storage_into(
-      const uint8_t *storage_key_data, uint32_t storage_key_length,
-      const uint8_t *key_data, uint32_t key_length, uint8_t *value_data,
-      uint32_t value_length, uint32_t value_offset) {
-    std::terminate();
+  runtime::SizeType StorageExtension::ext_storage_changes_root(
+      runtime::WasmPointer parent_hash_data, runtime::SizeType parent_hash_len,
+      runtime::SizeType parent_num, runtime::WasmPointer result) {
+    // TODO (kamilsa): PRE-95 Implement ext_storage_changes_root, 03.04.2019
+    logger_->error("Unimplemented");
+    return 0;
```

Comment:
Why not terminate?

Reply:
Same is done for [JS implementation](https://github.com/polkadot-js/client/blob/master/packages/client-runtime/src/storage/trie.ts#L31). I think it is okay for now assume no changes